src/tests/frontend/controller/test_sqlite_controller.py

```diff
@@ -65,24 +71,145 @@ def testSourcePathIfExisting(self):
+    expected_name = 'Parse{0}Row'.format(name)
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = os.path.join(tmpdir, 'testfile')
+      pathlib.Path(path).touch()
+
+      output_handler = output_handler_file.OutputHandlerFile(
+          path, file_handler.FileHandler(), prompt_info=name)
+      plugin_helper = fake_sqlite_plugin_helper.FakeSQLitePluginHelper(
+          folder_exists=True)
+      controller = sqlite_controller.SQLiteController(output_handler,
+                                                      plugin_helper)
+      actual = controller._CreateSQLQueryModelWithUserInput(sql_query, False,
+                                                            fake_execution)
+      self.assertIsNone(actual)
+
+  def testSqlQuery(self):
+    """test method after getting the source path from the user"""
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = os.path.join(tmpdir, 'testfile')
+      pathlib.Path(path).touch()
+
+      output_handler = output_handler_file.OutputHandlerFile(
+          path, file_handler.FileHandler(), confirm=False)
+      plugin_helper = fake_sqlite_plugin_helper.FakeSQLitePluginHelper()
+      controller = sqlite_controller.SQLiteController(output_handler,
+                                                      plugin_helper)
+
+      controller._CreateSQLQueryModelWithUserInput = mock.MagicMock(
+          return_value=base_sql_query_execution.SQLQueryData(
+              data='test', has_error=False, error_message=None))
+
+      actual = controller.SQLQuery(None, None, True)
+
+      prompt_output_actual = self._ReadFromFile(path)
+
+      prompt_output_expected = (
+        'Please write your SQL script for the pluginDo you want to add another '
+        'query?')
+
+      self.assertEqual(len(actual), 1)
+      self.assertEqual(actual[0].data, 'test')
+      self.assertEqual(actual[0].has_error, False)
+      self.assertEqual(actual[0].error_message, None)
+      self.assertEqual(prompt_output_actual, prompt_output_expected)
+
+  def testSqlQueryMultiple(self):
+    """test method after getting the source path from the user"""
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = os.path.join(tmpdir, 'testfile')
+      pathlib.Path(path).touch()
+
+      output_handler = output_handler_file.OutputHandlerFile(
+          path, file_handler.FileHandler(), confirm=True, confirm_amount_same=3)
+      plugin_helper = fake_sqlite_plugin_helper.FakeSQLitePluginHelper()
+      controller = sqlite_controller.SQLiteController(output_handler,
+                                                      plugin_helper)
+
+      controller._CreateSQLQueryModelWithUserInput = mock.MagicMock(
+          return_value=base_sql_query_execution.SQLQueryData(
+              data='test', has_error=False, error_message=None))
+
+      actual = controller.SQLQuery(None, None, True)
+
+      prompt_output_actual = self._ReadFromFile(path)
+
+      prompt_output_expected = '{0}{0}{0}'.format(
```

Comment:
Because i didn't want to write the same string 3 times.
But i could write 'the string'*3 instead.